datadryad/dryad-repo
# Dash transfer -- refactored version

This subsumes #1477, and refactors the essential pieces into three separate parts:
- DryadDataPackage, Author, and Package = serialization of a Dryad package into Dash-formatted JSON
- DashService = API communications with Dash
- TransferToDash = curation task that manages bulk transfer of packages to Dash

Also cleaned up Daisie's notes from #1477.... note that the names of the maven parameters have changed.

To test:
1. Ensure your maven settings file has appropriate values for 
    - `default.dash.server` = protocol and hostname of the target server (e.g. http://ryandash.datadryad.org)
    -  `default.dash.application.id` = oauth application ID
    - `default.dash.application.secret` = oauth application secret
2. Find the handle or DOI of an item you want to transfer
3. To post a new item to Dash, call the curation tool from the command line, using the handle:
`/opt/dryad/bin/dspace curate -v -t transfertodash -i 10255/dryad.135814 -r -`
4. To view an item from Dash, call the dash-service tool from the command line, using the DOI: 
`/opt/dryad/bin/dspace dash-service doi:10.5072/fzpe-zz40`

Note: Most new functionality will be added into the dash-service tool, but the curation tool is being retained since it will likely be better for bulk transfers.
